src/components/UserCards/UserCard.tsx

```diff
@@ -0,0 +1,211 @@
+import Rating from "@mui/material/Rating";
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+import timezone from "dayjs/plugin/timezone";
+import { ButtonInfo, EnhancedPublicUser, PublicUser } from "../../utils/types";
+import { trpc } from "../../utils/trpc";
+import { toast } from "react-toastify";
+import { useContext } from "react";
+import { UserContext } from "../../utils/userContext";
+import Spinner from "../Spinner";
+import { classNames } from "../../utils/classNames";
+import { User } from "@prisma/client";
+import { ConnectCard } from "./ConnectCard";
+import { SentCard } from "./SentCard";
+import { ReceivedCard } from "./ReceivedCard";
+
+interface UserCardProps {
+  otherUser: EnhancedPublicUser;
+  rightButton: ButtonInfo;
+  onViewRouteClick: (user: User, otherUser: PublicUser) => void;
+}
+
+export const renderUserCard = (
```

Comment:
My understanding is that this function's purpose is to render the correct type of user card (connect, received, or sent) based on the type of sidebar we are rendering (recommendations, favs, sent, or received). If that is the case, I feel that this function should be in the `SidebarContent` file.

Also, the `recommendations` case is empty. I believe it should render a `ConnectCard`.

Reply:
recommendations case is empty to allow fall through. That way it renders the same card for favorites and recommendations. I can add it back I was just trying to bring down number of lines of code

Reply:
I think it should be added, **or** the `subType` can be made into an enumeration of `connect`, `sent`, and `received` which makes more sense I guess.

Reply:
addressed the other comment